src/arrayMethodJoin.js

```diff
@@ -1,11 +1,31 @@
 'use strict';
 
-/**
- * Implement method join
- */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1) {
+        str += this[i];
+        break;
+      }
+
+      if (this[this.length - 1] === undefined) {
+        this[this.length - 1] = '';
```

Comment:
```suggestion
        this[this.length - 1] = '';
```
Метод join не мутує ніяким чином вхідний масив. Коли ти пишеш `this[i] = ` ти змінюєш вхідний масив. Давай цього уникнемо